compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml

```diff
@@ -624,10 +624,25 @@ null : [a] -> Bool
 null [] = True
 null _ = False
 
--- | Filter the list using the function: keep only the elements where the predicate holds.
+-- | Filters the list using the function: keep only the elements where the predicate holds.
 filter : (a -> Bool) -> [a] -> [a]
 filter p = foldr (\x xs -> if p x then x :: xs else xs) []
 
+-- | Filters the list using the applicative function: keeps only the elements where the predicate holds.
+-- Useful for example when a list of contract IDs are known, which should be filtered based on some
+-- condition for the correspodning contracts.
+--
+-- ```
+-- >>> filterA (\x -> Some (x % 2 == 0)) [0..5]
+-- Some [0, 2, 4]
+-- ```
+filterA : Applicative m => (a -> m Bool) -> [a] -> m [a]
```

Comment:
I think this function should be in `DA.Action` rather than `Prelude`?

Reply:
I thought it's similar in nature to `mapA` hence put here. I can move it.

Reply:
Yes, I see. I note that `mapM` is in the Haskell `Prelude` but `filterM` requires `Control.Monad`. If we are taking our queues from that decision then it probably belongs in `DA.Action` (along with `foldrA` and friends).